Register from all pages and add FAQ

anjelbenitez opened the pull request:
I've added the register feature to every page as well as a route to the FAQ page. For the login page, clicking on register will simply cause the register window to pop up and cover the login form.

Also, the login and register forms had the same input IDs for username and password, which caused an issue particularly in the login page where both logging in and registering needed to be implemented. To remedy this, I changed the input IDs of the username & password in the register form for every .handlebars file as well as in the register.js file.

anjelbenitez committed:
Added route for FAQ page

anjelbenitez committed:
Updated username/password with new input IDs

The login and register forms had the same input IDs for username and password, which caused an issue particularly in the login page where both logging in and registering needed to be implemented. To remedy this conflict, I changed the input IDs of the username & password in the register form. I also updated the handlebars file for every page.

anjelbenitez committed:
Updated to exclude register_close div

This particular css code was causing a misalignment for the "register_close" div of the registration window.

anjelbenitez committed:
Added register feature

anjelbenitez committed:
Create faq.handlebars

anjelbenitez committed:
Changed username/password input IDs

Changed username/password input IDs

anjelbenitez committed:
Added register functionality

anjelbenitez committed:
Added register functionality

anjelbenitez committed:
Added register functionality

anjelbenitez committed:
Added register functionality

alexlo510 committed:
register page can show on ingredient route

anthonyliuhao commented on main.js:
```diff
@@ -228,6 +235,7 @@ app.get('/ingredients/:recipename', function(req,res, next){
   context.user = req.user || null  // req.user exists when a user is logged in
   var recipe = req.params.recipename;
   context.title = "Ethical Eating - " + recipe;
+  context.js = ["register.js"]
```
You can simply add the line below to `ingredients.handlebars`.
```
<script src="js/register.js"></script>
```

anjelbenitez replied:
Thank you!

anthonyliuhao commented on views/build.handlebars:
```diff
@@ -1,3 +1,69 @@
+</div>
+<!--End Model-->
+
 <<<<<<< HEAD
```
Was there a conflict here? Please remove this line.

anjelbenitez replied:
I'm not sure what that was. I thought someone else put it there as a placeholder, but I will remove it now.

anthonyliuhao commented on views/breakfast.handlebars:
```diff
@@ -1,3 +1,69 @@
+<!--Model-->
+<div class="register_modal_container">
```
It bothers me that the same code block is in every handlebars file. Why don't we put this block in the `main.handlebars` layout template if we expect to need this on every page?

anjelbenitez replied:
Oh, I wasn't aware we could do that, but I will implement that as part of the code smells requirement.

anthonyliuhao commented on public/js/register.js:
```diff
@@ -15,8 +15,8 @@ function clearSignUpForm(){
     let email = document.getElementById('email');
-    let username = document.getElementById('username');
-    let password = document.getElementById('password');
+    let username = document.getElementById('user');
+    let password = document.getElementById('pass');
```
This may just be me nit-picking, but I am a little concerned about the approach to call the same things (`username` and `password`) different names just as a workaround. Can we figure out a way to not have the two conflicting HTML forms on the same page? For example, when user clicks on register on the login page, then just take them to a dedicated `/register` page.

anjelbenitez replied:
I don't think we have a dedicated register page, which is why that was the solution I went with. It seems like the register feature was implemented as a pop up window. I think it's fine to have them on the same page, if that's the case. The register window pops up when clicked and covers the login form, after which the user can close out the window and login after an account has been created.

The issue was that the register.js file was taking the input of "username" and "password" from the login form because the id's for the password and username input fields were the same for both forms. Changing the id for one fixed the issue.

anthonyliuhao replied:
Alright, let's leave it as is then.

anthonyliuhao reviewed (changes requested):
Please see the requested changes.

anjelbenitez committed:
Update build.handlebars

anjelbenitez committed:
Added register.js script to allow for register functionality

anjelbenitez commented:
closing to make changes

anjelbenitez committed:
fixed register.js path

anjelbenitez committed:
Removed redundant register HTML

anjelbenitez committed:
Removed redundant register HTML

anjelbenitez committed:
Removed redundant register HTML

anjelbenitez committed:
Removed redundant register HTML

anjelbenitez committed:
Removed redundant register HTML

anjelbenitez committed:
Removed redundant register HTML

anjelbenitez committed:
Added common register HTML

anjelbenitez committed:
Removed duplicate code

anjelbenitez committed:
Removed redundant register HTML

anjelbenitez committed:
Removed redundant register HTML

anjelbenitez committed:
Added z-index to login form container

This allows the login page to be positioned behind the register form in the login page if the register button were to be clicked.

anjelbenitez commented:
I added the register HTML into the main.handlebars file and removed redundant code. Also removed the ">>>>HEAD" text in the build.handlebars file. The register functionality now works for the individual "view ingredients" pages.

anthonyliuhao reviewed (approved):
Looks good to me! Thank you!
